lldb/source/Target/Process.cpp

```diff
@@ -3281,7 +3288,11 @@ Status Process::PrivateResume() {
             "Process::PrivateResume PreResumeActions failed, not resuming.");
       } else {
         m_mod_id.BumpResumeID();
-        error = DoResume();
+        if (direction == eRunForward) {
+          error = DoResume();
```

Comment:
What do we get out of the asymmetry between all the other resume methods that take a direction, and the doResume methods that don't?  It just looks awkward to my eyes.

Reply:
The advantage of separate `DoResume()` and `DoResumeReverse()` methods is that `Process` subclasses that don't support reverse execution don't have to change at all; they inherit the default `DoResumeReverse()` that returns an error. If I make `DoResume()` take a direction parameter then those implementations (three out of four AFAICT) need to add a check of the direction parameter. It's not a big deal, I'll change it.

Reply:
I've updated the branch with all these changes.